src/System.IO/src/System/IO/BufferedStream.cs

```diff
@@ -234,7 +236,11 @@ protected override void Dispose(bool disposing)
             finally
             {
                 _stream = null;
-                _buffer = null;
+                if (_buffer != null)
+                {
+                    ArrayPool<byte>.Shared.Return(_buffer);
```

Comment:
To clarify, I think we should probably just clear the buffers here to take the conservative route, but I'm also just trying to figure out how we are thinking about this buffer pooling stuff going forward, and what our best practices are going to be, etc.